ensight_lib/include/ensightbarycentriccoordinates.h

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2016 Fraunhofer ITWM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#ifndef ENSIGHTBARYCENTRICCOORDINATES_H
+#define ENSIGHTBARYCENTRICCOORDINATES_H
+
+#include <string>
+#include <QString>
+#include "eigentypes.h"
+
+// Forward declaration
+class EnsightCellIdentifier;
+
+/**
+ * @brief The EnsightBarycentricCoordinates class provides methods to compute barycentric coordinates in different cell types.
+ *
+ * Supported cell types are:  Ensight::Tetradhedron ,  Ensight::Pyramid ,  Ensight::Wedge ,  Ensight::Hexahedron.
+ */
+class EnsightBarycentricCoordinates
+{
+public:
+    EnsightBarycentricCoordinates();
```

Comment:
This could use =default if the cell_ used inline initialization

Reply:
The default constructor of Vec3 does not zero initialize, as far as I can tell. Anyways, I prefer the constructor to be out-of-line in the cpp file here, even if it were explicitly defaulted. This causes the code to be generated in the library instead of at calling site.